cypress/tests/widgets/chart-options.spec.js

```diff
@@ -0,0 +1,121 @@
+/// <reference types="cypress" />
+import should from 'should'
+
+/* eslint-disable cypress/no-unnecessary-waiting */
+describe('Node/-RED Dashboard 2.0 - Chart Widget - chart options', () => {
+    it('resets tooltip mode to "nearest" for bar charts', () => {
+        const chartName = 'bar-chart-1'
+        const overrides = [
+            {
+                id: 'chart-1', // the ID in the base flow fixture (required)
+                mode: 'merge',
+                data: {
+                    label: chartName, // set the label
+                    chartType: 'bar' // change the chart type to bar
+                }
+            }
+        ]
+        cy.deployFixture('dashboard-chart-options', overrides)
+        cy.visit('/dashboard/page1')
+        cy.reload()
+
+        cy.get('#nrdb-ui-widget-chart-1 > div > canvas').should('exist')
+        cy.clickAndWait(cy.get('button').contains('random'), 10)
+
+        // eslint-disable-next-line promise/catch-or-return, promise/always-return
+        cy.window().then(win => {
+            should(win.uiCharts).is.not.empty()
+            const barChart = win.uiCharts['chart-1']
+            should(barChart).is.not.empty()
+            should(barChart.chart.options.interaction).be.an.Object()
+            should(barChart.chart.options.interaction.mode).be.equal('nearest')
+            should(barChart.chart.options.interaction.axis).not.equal('x')
+        })
+    })
+
+    it('sets tooltip mode to "index" and axis x for line charts', () => {
+        const chartName = 'line-chart-1'
+        const overrides = [
+            {
+                id: 'chart-1', // the ID in the base flow fixture (required)
+                mode: 'merge',
+                data: {
+                    label: chartName, // set the label
+                    chartType: 'line' // change the chart type to line
+                }
+            }
+        ]
+        cy.deployFixture('dashboard-chart-options', overrides)
+        cy.visit('/dashboard/page1')
+        cy.reload()
+
+        cy.get('#nrdb-ui-widget-chart-1 > div > canvas').should('exist')
+        cy.clickAndWait(cy.get('button').contains('random'), 10)
+
+        // eslint-disable-next-line promise/catch-or-return, promise/always-return
+        cy.window().then(win => {
+            should(win.uiCharts).is.not.empty()
```

Comment:
If `win.uiCharts` is already available, then why also the need for `window.uiChart` which is commented out?

Reply:
Not sure I understand the question Joe.

Where is window.uiChart commented out?

For reference: `win.uiCharts` is added explicitly in `UIChart.vue` when `Cypress` is detected (only). 
This is pretty much the only way of getting to the end for E2E tests. It was debated and researched with Serban at the time of impleneting the first round of e2e tests for charts.

Reply:
> window.uiChart which is commented out?

ah, you mean in the other file - in UIChart.vue:
https://github.com/FlowFuse/node-red-dashboard/pull/1336/files/6040f1ca6c988150cb901e0d1292552e7e925614

Those commented out lines were super helpful during development of tests (so I could see what was set in the chart  object).

the line you have commented on here is related directly to existing code here
https://github.com/FlowFuse/node-red-dashboard/pull/1336/files/6040f1ca6c988150cb901e0d1292552e7e925614#diff-0ab94e22f1954c12fd38be2754bb91b4c72e3ad50b23742c57a12ff9c3e9e194R60-R65

As mentioned above, `uiChart` is conditionally added to the window object ONLY when cypress is detected. This highlighted code (in the test runner) picks up the chart object to get as close to the end as possible.